hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/SerializationServiceV1.java

```diff
@@ -137,25 +137,24 @@ protected void registerConstantSerializers() {
         registerConstant(String[].class, new StringArraySerializer());
     }
 
-    public void registerJavaTypeSerializers(boolean enableCompression, boolean enableSharedObject) {
-        safeRegister(Date.class, new DateSerializer());
-        safeRegister(BigInteger.class, new BigIntegerSerializer());
-        safeRegister(BigDecimal.class, new BigDecimalSerializer());
-        safeRegister(Externalizable.class, new ExternalizableSerializer(enableCompression));
-        safeRegister(Serializable.class, new JavaSerializer(enableSharedObject, enableCompression));
-        safeRegister(Class.class, new ClassSerializer());
-        safeRegister(Enum.class, new EnumSerializer());
-        safeRegister(ArrayList.class, new ArrayListStreamSerializer());
-        safeRegister(LinkedList.class, new LinkedListStreamSerializer());
+    public void registerJavaTypeSerializers() {
+        //Java extensions: more serializers
+        registerConstant(Date.class, new DateSerializer());
+        registerConstant(BigInteger.class, new BigIntegerSerializer());
+        registerConstant(BigDecimal.class, new BigDecimalSerializer());
+        registerConstant(Class.class, new ClassSerializer());
+        registerConstant(Enum.class, new EnumSerializer());
+        registerConstant(ArrayList.class, new ArrayListStreamSerializer());
+        registerConstant(LinkedList.class, new LinkedListStreamSerializer());
+
+        safeRegister(Serializable.class, javaSerializerAdapter);
```

Comment:
is this necessary to register Serializable? We already check if a class is Serializable in the lookupJavaSerializer, and means you can also skip removing the two interfaces in the lookupCustomSerializer